haiku/_src/typing.py

```diff
@@ -36,6 +36,7 @@
 State = Mapping[str, Mapping[str, jnp.ndarray]]
 Padding = Callable[[int], Sequence[int]]
 Paddings = Union[Padding, Sequence[Padding]]
+Regularizer = Callable[[jnp.ndarray], float]
```

Comment:
This takes any tree of ndarrays and returns an ndarray, I think the type should be `Callable[[Any], jnp.ndarray]` or `Callable[[Params], jnp.ndarray]`.

Reply:
Yeah, good point!